nautobot/extras/management/commands/fix_custom_fields.py

```diff
@@ -0,0 +1,40 @@
+from logging import getLogger
+
+from django.contrib.contenttypes.models import ContentType
+from django.core.management.base import BaseCommand
+from django.db import transaction
+
+from nautobot.extras.models import CustomField
+from nautobot.extras.utils import FeatureQuery
+
+logger = getLogger("nautobot.extras.fix_custom_fields")
+
+
+class Command(BaseCommand):
+    help = "Reset custom field attachments"
+
+    def handle(self, *args, **kwargs):
+        """Run through all objects and ensure they are associated with the correct custom fields."""
+        content_types = ContentType.objects.filter(FeatureQuery("custom_fields").get_query())
+
+        for content_type in content_types:
+            logger.info(f"Processing ContentType {content_type}")
+            model = content_type.model_class()
+            custom_fields_for_content_type = content_type.custom_fields.all()
+            custom_field_names_for_content_type = [cf.name for cf in custom_fields_for_content_type]
+            with transaction.atomic():
+                for obj in model.objects.all():
+                    obj_changed = False
+                    # Provision CustomFields that are not associated with the object
+                    for custom_field in custom_fields_for_content_type:
+                        if not obj._custom_field_data[custom_field.name]:
+                            logger.debug(f"Adding CustomField {custom_field.name} to {obj}")
+                            obj._custom_field_data[custom_field.name] = custom_field.default
+                            obj_changed = True
+                    # Remove any custom fields that are not associated with the content type
+                    for field_name in set(obj._custom_field_data) - set(custom_field_names_for_content_type):
+                        logger.debug(f"Removing CustomField {field_name} from {obj}")
+                        del obj._custom_field_data[field_name]
+                        obj_changed = True
+                    if obj_changed:
+                        obj.validated_save()
```

Comment:
I think we should add a try/except block to catch any ValidationError, otherwise a bad object that isn't currently valid could stop the execution of the script prematurely and prevent the script to fix all CFs